lib/rules/jsx-max-depth.js

```diff
@@ -91,7 +91,7 @@ module.exports = {
     function findJSXElementOrFragment(startNode, name, previousReferences) {
       function find(refs, prevRefs) {
         for (let i = refs.length - 1; i >= 0; i--) {
-          if (has(refs[i], 'writeExpr')) {
+          if (has(refs[i], 'writeExpr') && refs[i].writeExpr) {
```

Comment:
when is the property present but falsy?

Reply:
This is the output of keys/values of the refs:

|  v7  |  v8  |
| --- | ---- |
|   <img width="324" alt="Screenshot 2024-09-06 at 23 33 42" src="https://github.com/user-attachments/assets/27db0ff8-16a3-48a0-8d8b-a93dd553978d">   |   <img width="330" alt="Screenshot 2024-09-06 at 23 33 55" src="https://github.com/user-attachments/assets/d8a99123-0102-4d8d-96e9-c2f2b57d3b89">   |

The source code doesn't look a whole lot different, so maybe there's some changes between how the code is built before publishing 🤔

Reply:
I'm confused, what value does it hold in the v8 image in the bottom?

Reply:
Sorry, that was the terminal color scheme 🤦  They are `undefined`

<img width="326" alt="Screenshot 2024-09-07 at 00 12 50" src="https://github.com/user-attachments/assets/050db89c-16a0-4290-a8bc-9291c5178d57">

Reply:
so it's not about truthy/falsy, just undefined? in that case instead of checking `has` can we just check if `typeof refs[i].writeExpr === 'undefined'`?

Reply:
We sure can 🙂 I updated the commit and no issues in the CI 👍